domain/storageprovisioning/state/base_test.go

```diff
@@ -193,6 +193,41 @@ VALUES (?, ?, 0, 0)
 	return fsUUID, fsID
 }
 
+// newModelFilesystemAttachmentWithMount creates a new filesystem attachment
+// that has model provision scope. The attachment is associated with the
+// provided filesystem uuid and net node uuid. This will also set the mount
+// point and readonly attributes of the filesystem attachment.
+func (s *baseSuite) newModelFilesystemAttachmentWithMount(
+	c *tc.C,
+	fsUUID storageprovisioning.FilesystemUUID,
+	netNodeUUID domainnetwork.NetNodeUUID,
+	mountPoint string,
+	readOnly bool,
+) storageprovisioning.FilesystemAttachmentUUID {
+	attachmentUUID := domaintesting.GenFilesystemAttachmentUUID(c)
+	_, err := s.DB().ExecContext(
```

Comment:
This should be a transaction.

Reply:
The database can become locked, it just makes the tests more flaky.

We should get someone to go through them all and ensure we use the transaction runner. Can be done independently of this PR.